[DOCU-2187] Add config.canary_by_header_name to canary plugin (#3673)
Kong/docs.konghq.com: app/_hub/kong-inc/canary/index.md

kong_version_compatibility:
  enterprise_edition:
    compatible:
      - 2.8.x
      - 2.7.x
      - 2.6.x

>       - 2.5.x

Reviewer: Should remove the previous versions, if version 0.6.x comes with gateway 2.8.x.

      datatype: string
      description: |
        Header name whose value will be used as hash input. Required if `config.hash` is set to `header`.
    - name: canary_by_header_name
      required: null
      default: null
      value_in_examples: null
      datatype: string
      description: |

>         Header name that, when present on a request, will override the configured canary functionality. If the configured header is present with the value `always` then the request will always go to the canary upstream. If the header is present with the value `never` then the request will never go to the canary upstream. For all other values, the configured canary rules will be applied.

Reviewer: ```suggestion
        Header name that, when present on a request, overrides the configured canary 
        functionality. 
        
        * If the configured header is present with the value `always`, the request will 
        always go to the canary upstream. 
        * If the header is present with the value `never`, the request will never go to the 
        canary upstream. 
        
        For all other values, the configured canary rules will be applied.
```

Canary provides an automatic fallback if, for some reason, a Consumer, Header, or IP can
not be identified. The fallback order is `consumer`->`ip`->`none`.

### Overriding the Canary

> In some cases, it may be desirable to allow clients to pick _Service A_ or _Service B_

Reviewer: ```suggestion
In some cases, you may want to allow clients to pick _Service A_ or _Service B_
```
